### PR TITLE
Add all circle ci tests to testgrid

### PR DESCRIPTION
**Pull request by chxchx**
Results are up at https://k8s-testgrid.appspot.com/istio#circleci-e2e-pilot but contains only the runs from this particular PR. Once it is merged all the test results will show up.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -511,14 +493,18 @@ jobs:
     resource_class: xlarge
     steps:
       - checkout
+      - <<: *markJobStartsOnGCS
       - run:
           command: |
-            mkdir -p /go/out/racetests
-            trap "go-junit-report </go/out/racetests/go-racetest-report.out > /go/out/racetests/go-racetest-report.xml" EXIT
+            mkdir -p /go/out/tests
```
Comment (sebastienvas) [inline .circleci/config.yml]: check with @xiaolanz if this is OK.

Comment (chxchx) [reply .circleci/config.yml]: Yes, just did, and she is happy with it.

**Comment by hklai** [comment]
Merging this to get better circleci test visibility on testgrid.

**Pull request merged**